### PR TITLE
Add a function that extracts zipped folders of solidity contracts and compiles them

### PR DESCRIPTION
**Pull request by arajasek**
We currently do not support compiling Solidity contracts that import other contracts, as described here: aionnetwork/aion#611.

This PR makes the changes to the FVM that allows it to extract zipped folders, compile the contracts in them, and clean up after itself.

### CODE REVIEW

File: modFastVM/src/org/aion/solidity/Compiler.java [hunk 1]
```diff
@@ -39,6 +43,37 @@ public Result compile(byte[] source, Options... options) throws IOException {
...
+            boolean combinedJson,
+            Options... options)
+            throws IOException {
+        File unzipped = extractZip(source, "temp");
+
+        List<String> commandParts = prepareCommands(optimize, combinedJson, options);
+        commandParts.add(entryPoint);
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commandParts).directory(unzipped);
+        return runCompileProcess(source, processBuilder);
```
Comment (yulongaion) [inline modFastVM/src/org/aion/solidity/Compiler.java]: I don't think we need the `source` bytes to compile files in the temp folder.

File: modFastVM/src/org/aion/solidity/Compiler.java [hunk 2]
```diff
@@ -103,6 +118,27 @@ public Result compile(byte[] source, boolean optimize, boolean combinedJson, Opt
         }
     }
 
+    private File extractZip(byte[] source, String tempDirName) throws IOException {
```
Comment (AionJayT) [inline modFastVM/src/org/aion/solidity/Compiler.java]: Should check the nullity of the input args cause you only catch the IOException

File: modFastVM/src/org/aion/solidity/Compiler.java [hunk 3]
```diff
@@ -103,6 +118,27 @@ public Result compile(byte[] source, boolean optimize, boolean combinedJson, Opt
...
+        File tempDir = new File(tempDirName);
+        tempDir.mkdir();
```
Comment (AionJayT) [inline modFastVM/src/org/aion/solidity/Compiler.java]: should check the directory has been created. then go next step.

File: modFastVM/src/org/aion/solidity/Compiler.java [hunk 4]
```diff
@@ -61,6 +43,37 @@ public Result compile(byte[] source, Options... options) throws IOException {
...
+            boolean combinedJson,
+            Options... options)
+            throws IOException {
+        File unzipped = extractZip(source, "temp");
+
+        List<String> commandParts = prepareCommands(optimize, combinedJson, options);
+        commandParts.add(entryPoint);
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commandParts).directory(unzipped);
+        return runCompileProcess(source, processBuilder);
```
Comment (AionJayT) [inline modFastVM/src/org/aion/solidity/Compiler.java]: after compile finished. Should delete the tmp folder?

File: modFastVM/src/org/aion/solidity/Compiler.java (hunk 3 above)
Comment (arajasek) [reply modFastVM/src/org/aion/solidity/Compiler.java]: Fixed

**Pull request merged**